dandiapi/api/permissions.py

```diff
@@ -0,0 +1,29 @@
+from rest_framework.permissions import SAFE_METHODS, BasePermission, IsAuthenticated
+
+from dandiapi.api.models.user import UserMetadata
+
+
+class IsApproved(IsAuthenticated):
+    def has_permission(self, request, view):
+        return bool(
+            super().has_permission(request, view)
+            and (
+                request.user.is_superuser
+                or request.user.metadata.status == UserMetadata.Status.APPROVED
```

Comment:
Note that this will result in an extra SQL query for every request that hits an endpoint protected by these permission classes. It should be possible to override the current auth backend to use `select_related` to mitigate this. This PR is already massive enough though, so this can postponed to a future PR if we want to do it.